store/src/java/com/zimbra/cs/db/DbMailItem.java

```diff
@@ -1157,6 +1157,34 @@ public void update(MailItem item, Metadata metadata) throws ServiceException {
             DbPool.closeStatement(stmt);
         }
     }
+    
+    public void updateSharedFile(MailItem item, Metadata metadata, String fileName, Long date, Long size) throws ServiceException {
+        String name = fileName.isEmpty() ? item.getName() : fileName;
+        checkNamingConstraint(mailbox, item.getFolderId(), name, item.getId());
+
+        DbConnection conn = mailbox.getOperationConnection();
+        PreparedStatement stmt = null;
+        try {
+            stmt = conn.prepareStatement("UPDATE " + getMailItemTableName(item)
+                    + " SET  date = ?, size = ? , name = ? , mod_metadata = ?, "
+                    + "  change_date = ?, mod_content = ?" + " WHERE " + IN_THIS_MAILBOX_AND + "id = ?");
+            int pos = 1;
+            stmt.setInt(pos++, (int) (date / 1000));
```

Comment:
Why is this divide by 1000?

Reply:
discussed on call

Reply:
can we use `Instant.now().getEpochSecond();` instead please confirm.

Reply:
Since dates has to be in sync and we are using the exact date/epochdate at which file was modified
in this way file date at sharer end and receiver end remains same
so I would refrain from using this at the moment date , as there might be discrepancies in time